File: backend/src/main/java/com/demo/student_management/dto/admin/TeacherUpdateRequest.java
package com.demo.student_management.dto.admin;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class TeacherUpdateRequest {
    private String ten;
    private String matKhau; // optional, only updated if not empty/null
    private String gioiTinh;
    private String email;
    private String sdt;
    private String idLop;
    private String idMonHoc;
    private Boolean active;
}

File: frontend/StudentManagement.Shared/Dtos/Admin/TeacherUpdateRequest.cs
namespace StudentManagement.Shared.Dtos.Admin;

public sealed class TeacherUpdateRequest
{
    public string Ten { get; set; } = string.Empty;
    public string MatKhau { get; set; } = string.Empty;
    public string GioiTinh { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string Sdt { get; set; } = string.Empty;
    public string IdLop { get; set; } = string.Empty;
    public string IdLopChuNhiem { get; set; } = string.Empty;
    public string IdMonHoc { get; set; } = string.Empty;
    public bool Active { get; set; } = true;
}
